### PR TITLE
Add JS (lib/) subsystem labels

### PR DESCRIPTION
**Pull request by Fishrock123**
Depends on https://github.com/nodejs-github-bot/github-bot/pull/41

Does some.. fancy.. mapping for some of the outliers.

Also limits the labels to 4, and just uses `lib / src` at 5.

cc @phillipj

### CODE REVIEW

File: lib/node-labels.js [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
+  // Oddities first
+  [/^lib\/string_decoder/, 'buffer'],
+  [/^lib\/sys\.js$/, 'util'],
+  [/^lib\/constants\.js$/, 'lib / src'],
+  [/^lib\/_debug_agent\.js$/, 'debugger'],
+  [/^lib\/punycode|^lib\/\w+\/freelist/, ''], // TODO: ignore better?
+  [/^lib(\/\w+)?\/(_)?link(ed)?list/, 'timers'],
+  [/^lib\/\w+\/bootstrap_node/, 'lib / src'],
+  [/^lib\/\w+\/v8_prof_/, 'tools'],
+  [/^lib\/\w+\/socket_list/, 'net'],
```
Comment (Fishrock123) [inline lib/node-labels.js]: `\/\w+` in some of these is just a shorter `\/internal`

**Comment by Fishrock123** [comment]
Rebased on `master`.

File: lib/node-labels.js [hunk 2]
```diff
@@ -9,24 +9,52 @@ const subSystemLabelsMap = new Map([
...
+
+  /* Dependencies */
   // libuv needs an explicit mapping, as the ordinary /deps/ mapping below would
   // end up as libuv changes labeled with "uv" (which is a non-existing label)
   [/^deps\/uv\//, 'libuv'],
-  [/^deps\/([^/]+)/, '$1']
+  [/^deps\/([^/]+)/, '$1'],
+
+  /* JS subsystems */
+  // Oddities first
```
Comment (phillipj) [inline lib/node-labels.js]: Ohmy the oddities list was bigger than I imagined

**Comment by phillipj** [comment]
LGTM 👍

**Comment by Fishrock123** [comment]
Updated a bit, condensed some of the oddities a bit.

**Comment by phillipj** [comment]
@Fishrock123 deployed

We're now running v6 in production 👯

**Pull request merged**